src/pyhf/interpolators/code0.py

```diff
@@ -57,7 +57,7 @@ def __call__(self, alphasets):
         tensorlib, _ = get_backend()
         self._precompute_alphasets(tensorlib.shape(alphasets))
         where_alphasets_positive = tensorlib.where(
-            alphasets > 0, self.mask_on, self.mask_off
+            tensorlib.astensor(alphasets > 0), self.mask_on, self.mask_off
```

Comment:
why are we *adding* `astensor` calls here?

Reply:
In short, because I removed the `mask = self.astensor(mask)` from the `where` method in the TensorFlow backend. As it requires that the `mask` argument is a tensor I decided to add `tensorlib.astensor` calls to a few places where the mask is a logical requirement and not already a tensor given that

```
$ git grep "tensorlib.where(" src/ | wc -l
26
```

is more then the number of calls that I added. So it seems that explicitly calling it as needed is preferable then having the backends call it everytime.

Reply:
Alternatively, I could say that as this is a TensorFlow problem which might improve in TensorFlow 2.0 I could remove the calls to `tensorlib.astensor` that I added and just have TensorFlow do the redundant calls, until it might be possible to turn it off entirely.

Reply:
could we skip this if we instead assured that alphasets is a tensor to start with?

Reply:
> could we skip this if we instead assured that alphasets is a tensor to start with?

No. The issue is that `alphasets > 0` is being passed in. TensorFlow 1.0 needs to be told that this is a bool tensor with a `astensor` call.

Reply:
The decision is if we're going to make a few calls across all backends (which we are currently already doing) in the interpolator codes and hypotest, or if we are going to isolate all of those calls to the TensorFlow backend and deal with them later in the TensorFlow 2.0 migration.

I think we can probably do the latter.